### DISCUSSION TITLE
Diode parameter relationship between forward voltage and activation energy

### DISCUSSION THREAD

**Discussion by rede676** [SpiceSharp/SpiceSharp#215] [General]
I am looking to set the forward voltage @ 1A for a diode in a simulation ie. 2V @ 1A.
If I had a 5V power source, 1K resister and 2V@1A diode in series what I would expect is the Vd over the resister to be 3V and the Vd of the diode to be 2V.

is this achieved using activation energy and is there an equation that will turn forward voltage into eV the unit of  activation energy

**Comment by svenboulanger** [comment]
You can find the models from the source code here:
- [Diode biasing model](https://github.com/SpiceSharp/SpiceSharp/blob/9259fd2e81e9216c4653cdc380cdc43da89e3a2f/SpiceSharp/Components/Semiconductors/Diodes/Biasing.cs#L102)
- [Diode temperature-dependent calculations](https://github.com/SpiceSharp/SpiceSharp/blob/9259fd2e81e9216c4653cdc380cdc43da89e3a2f/SpiceSharp/Components/Semiconductors/Diodes/Temperature.cs#L136)
- [Diode model temperature-dependent calculations](https://github.com/SpiceSharp/SpiceSharp/blob/9259fd2e81e9216c4653cdc380cdc43da89e3a2f/SpiceSharp/Components/Semiconductors/Diodes/ModelTemperature.cs#L82)

I think those are sufficient for DC biasing conditions. If you ignore temperature-related and area scaling effects, then the formula is

```
Vt = Constants.KOverQ * Parameters.Temperature;
Vte = ModelParameters.EmissionCoefficient * Vt;
i = ModelParameters.SaturationCurrent * (exp(v / Vte) - 1)
```

You should be able to calculate the required $I_{sat}$ = `SaturationCurrent` and $\eta$ = `EmissionCoefficient` from these equations.
If you simply work with a emission coefficient of 1, then you will find

$$I_{sat}= \frac{1A}{e^\frac{2V}{U_{th}} - 1} \text{ with } U_{th} = \frac{kT}{q}$$

If your values are for room temperature, then this is approx. $I_{sat} \approx 8.86\cdot 10^{-35} A$.
